tunnel_virtual_networks.go

```diff
@@ -0,0 +1,157 @@
+	Comment   string `json:"comment,omitempty"`
+	IsDefault *bool  `json:"is_default,omitempty"`
+}
+
+type TunnelVirtualNetworkUpdateParams struct {
+	AccountID        string `json:"-"`
+	VnetID           string `json:"-"`
+	VnetName         string `json:"vnet_name,omitempty"`
+	Comment          string `json:"comment,omitempty"`
+	IsDefaultNetwork *bool  `json:"is_default_network,omitempty"`
+}
+
+type TunnelVirtualNetworkDeleteParams struct {
+	AccountID string `json:"-"`
+	VnetID    string `json:"-"`
+}
+
+// tunnelRouteListResponse is the API response for listing tunnel virtual networks.
+type tunnelVirtualNetworkListResponse struct {
+	Response
+	Result []TunnelVirtualNetwork `json:"result"`
+}
+
+type tunnelVirtualNetworkResponse struct {
+	Response
+	Result TunnelVirtualNetwork `json:"result"`
+}
+
+// ListTunnelVirtualNetworks lists all defined virtual networks for tunnels in the account.
+//
+// See: https://api.cloudflare.com/#tunnel-virtual-network-list-virtual-networks
+func (api *API) ListTunnelVirtualNetworks(ctx context.Context, params TunnelVirtualNetworksListParams) ([]TunnelVirtualNetwork, error) {
+	if params.AccountID == "" {
+		return []TunnelVirtualNetwork{}, ErrMissingAccountID
+	}
+
+	uri := buildURI(fmt.Sprintf("/%s/%s/teamnet/virtual_networks", AccountRouteRoot, params.AccountID), params)
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, params)
+	if err != nil {
+		return []TunnelVirtualNetwork{}, err
+	}
+
+	var resp tunnelVirtualNetworkListResponse
+	err = json.Unmarshal(res, &resp)
+	if err != nil {
+		return []TunnelVirtualNetwork{}, errors.Wrap(err, errUnmarshalError)
+	}
+
+	return resp.Result, nil
+}
+
+// CreateTunnelVirtualNetwork adds a new virtual network to the account
+//
+// See: https://api.cloudflare.com/#tunnel-virtual-network-create-virtual-network
+func (api *API) CreateTunnelVirtualNetwork(ctx context.Context, params TunnelVirtualNetworkCreateParams) (TunnelVirtualNetwork, error) {
+	if params.AccountID == "" {
+		return TunnelVirtualNetwork{}, ErrMissingAccountID
+	}
+
+	uri := fmt.Sprintf("/%s/%s/teamnet/virtual_networks", AccountRouteRoot, params.AccountID)
+
+	responseBody, err := api.makeRequestContext(ctx, http.MethodPost, uri, params)
+	if err != nil {
+		return TunnelVirtualNetwork{}, err
+	}
+
+	var resp tunnelVirtualNetworkResponse
+	err = json.Unmarshal(responseBody, &resp)
+	if err != nil {
+		return TunnelVirtualNetwork{}, errors.Wrap(err, errUnmarshalError)
+	}
+
+	return resp.Result, nil
+}
+
+// DeleteTunnelVirtualNetwork deletes an existing virtual network from the account.
+//
+// See: https://api.cloudflare.com/#tunnel-virtual-network-delete-virtual-network
+func (api *API) DeleteTunnelVirtualNetwork(ctx context.Context, params TunnelVirtualNetworkDeleteParams) error {
+	if params.AccountID == "" {
+		return ErrMissingAccountID
+	}
+
+	uri := fmt.Sprintf("/%s/%s/teamnet/virtual_networks/%s", AccountRouteRoot, params.AccountID, url.PathEscape(params.VnetID))
```

Comment:
what scenarios do we need `url.PathEscape` for here?

Reply:
It is redundant. Removed.